lib/sync-DataSetModel.js

```diff
@@ -259,22 +259,28 @@ var self = {
 
     self.getDataset(dataset_id, function (err, dataset) {
       if (err) return err;
-      var clientHash = self.getClientHash(query_params, meta_data);
-      var datasetClient = dataset.clients[clientHash];
-      if (!datasetClient) {
-        self.createDatasetClient(dataset_id, query_params, meta_data, function (err, dsc) {
-          if (err) return cb(err);
-          recordAccessInfo(dsc, params);
-          return cb(null, dsc);
-        });
-      }
-      else {
-        recordAccessInfo(datasetClient, params);
-        return cb(null, datasetClient);
-      }
+      //var clientHash = self.getClientHash(query_params, meta_data);
+      //console.log('clientHash - Sync Result : ' + clientHash);
+
+      self.asyncGetClientHash(query_params, meta_data, function(err, clientHash) {
+        console.log('clientHash - Async Result : ' + clientHash);
+        var datasetClient = dataset.clients[clientHash];
+        if (!datasetClient) {
+          self.createDatasetClient(dataset_id, query_params, meta_data, function (err, dsc) {
+            if (err) return cb(err);
+            recordAccessInfo(dsc, params);
+            return cb(null, dsc);
+          });
+        }
+        else {
+          recordAccessInfo(datasetClient, params);
+          return cb(null, datasetClient);
+        }
+      });
     });
 
     function recordAccessInfo(datasetClient, params) {
+      console.log('recordAccessInfo', params);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this logging statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS2228)